nomad/structs/identity.go

```diff
@@ -93,6 +149,31 @@ func (i *IdentityClaims) setNodeSubject(node *Node, region string) {
 	}, ":")
 }
 
+// setNodeSubject sets the "subject" or "sub" claim for the node introduction
+// identity JWT. It follows the format
+// "node-introduction:<region>:<node_pool>:<node_name>:default", where "default"
+// indicates identity name. While this is currently hardcoded, it could be
+// configurable in the future as we expand the node identity offering and allow
+// greater control of node access.If the operator does not provide a node name,
+// this is omitted from the subject.
```

Comment:
In the case where the node name is omitted, maybe we should use `::` in order to disambiguate the final field? ex. `node-introduction:global:foo::default` would be easier for consumers to parse into fields.